pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -946,6 +947,37 @@ impl DeltaLayerInner {
         Ok(planner.finish())
     }
 
+    fn get_min_read_buffer_size(
+        planned_reads: &[VectoredRead],
+        read_size_soft_max: usize,
+    ) -> usize {
+        let largest_read = match planned_reads.iter().max_by_key(|read| read.size()) {
+            Some(largest_read) => largest_read,
+            None => {
+                return read_size_soft_max;
+            }
+        };
+
+        let largest_read_size = largest_read.size();
+        if largest_read_size > read_size_soft_max {
+            // If the read is oversized, it should only contain one key.
+            let offenders = largest_read
+                .blobs_at
+                .as_slice()
+                .iter()
+                .map(|(_, blob_meta)| format!("{}@{}", blob_meta.key, blob_meta.lsn))
+                .join(", ");
+            tracing::warn!(
```

Comment:
What bounds the length of `offenders`?  We wouldn't want a warn log line with e.g. hundreds of keys in the worst case.  Might be fine just to pick the first key in the list as an example?

Reply:
Good question! It's the logic in `VectoredReadBuilder` & `VectoredReadPlanner`. If adding a new blob would violate the max read size, we push it to the next read. If the blob itself is larger than the read size, then it gets a read for itself. Since the read is already oversized, we will not add anything to it. QED. This is the only case when a read can go over the prescribed size.

Even if i'm wrong the max number of keys in a vectored read 32, so that's our worst case.